lms/services/event.py

```diff
@@ -38,6 +43,19 @@ def insert_event(self, event: BaseEvent):
 
         return event
 
+    @staticmethod
+    def queue_event(event: BaseEvent) -> None:
```

Comment:
This method abstracts the error handling and the whole queing of the task.

In doing so it introduces circular dependency task-service. See a non-on-the-top import on the task side.